cmd/nri-kubernetes/main.go

```diff
@@ -49,59 +48,92 @@ func main() {
 		os.Exit(ExitIntegration)
 	}
 
-	// TODO: Here we will switch-case between components: KSM, ControlPlane, etc.
-	err = runKSM(&conf, clients, i)
-	if err != nil {
-		logger.Errorf(err.Error())
-		os.Exit(ExitLoop)
+	var ksmScraper *ksm.Scraper
+	if c.KSM.Enabled {
+		ksmScraper, err = setupKSM(c, clients)
+		if err != nil {
+			logger.Errorf("setting up ksm scraper: %w", err)
+			os.Exit(ExitSetup)
+		}
+		defer ksmScraper.Close()
+	}
+
+	for {
+		// TODO think carefully to the signature of this function
+		err := runScrapers(c, ksmScraper, i, clients)
+		if err != nil {
+			logger.Errorf("retrieving scraper data: %v", err)
+			os.Exit(ExitLoop)
+		}
+
+		err = i.Publish()
+		if err != nil {
+			logger.Errorf("publishing integration: %v", err)
+			os.Exit(ExitLoop)
+		}
+
+		time.Sleep(c.Interval)
 	}
 }
 
-func buildClients() (*clusterClients, error) {
-	k8s, err := client.NewKubernetes(true)
-	if err != nil {
-		return nil, fmt.Errorf("building kubernetes client: %w", err)
+func runScrapers(c config.Mock, ksmScraper *ksm.Scraper, i *integration.Integration, clients *clusterClients) error {
+	if c.KSM.Enabled {
+		err := ksmScraper.Run(i)
+		if err != nil {
+			return fmt.Errorf("retrieving ksm data: %w", err)
+		}
 	}
 
-	ksmCli, err := ksmClient.New(ksmClient.WithLogger(logger))
-	if err != nil {
-		return nil, fmt.Errorf("building KSM client: %w", err)
+	if c.Kubelet.Enabled {
+		// TODO this is merely a stub running old code
+		err := deprecated.RunKubelet(&c, clients.k8s, i)
+		if err != nil {
+			return fmt.Errorf("retrieving kubelet data: %w", err)
+		}
 	}
 
-	return &clusterClients{
-		k8s: k8s,
-		ksm: ksmCli,
-	}, nil
+	if c.ControlPlane.Enabled {
+		// TODO this is merely a stub running old code
+		err := deprecated.RunControlPlane(&c, clients.k8s, i)
+		if err != nil {
+			return fmt.Errorf("retrieving control-plane data: %w", err)
+
+		}
+	}
+
+	return nil
 }
 
-func runKSM(config *config.Mock, clients *clusterClients, i *integration.Integration) error {
-	ksmScraper, err := ksm.NewScraper(config, ksm.Providers{
+func setupKSM(c config.Mock, clients *clusterClients) (*ksm.Scraper, error) {
+	providers := ksm.Providers{
 		// TODO: Get rid of custom client.Kubernetes wrapper and use kubernetes.Interface directly.
 		K8s: clients.k8s.GetClient(),
 		KSM: clients.ksm,
-	},
-		ksm.WithLogger(logger),
-	)
+	}
 
+	ksmScraper, err := ksm.NewScraper(&c, providers, ksm.WithLogger(logger))
 	if err != nil {
-		return fmt.Errorf("building KSM scraper: %w", err)
+		return nil, fmt.Errorf("building KSM scraper: %w", err)
 	}
 
-	defer ksmScraper.Close()
-
-	for {
-		err = ksmScraper.Run(i)
-		if err != nil {
-			return fmt.Errorf("scraping KSM: %w", err)
-		}
+	return ksmScraper, nil
+}
 
-		err = i.Publish()
-		if err != nil {
-			return fmt.Errorf("publishing integration: %w", err)
-		}
+func buildClients() (*clusterClients, error) {
+	k8s, err := client.NewKubernetes(true)
```

Comment:
Will we get rid of the custom Kubernetes interface in this PR, or not yet? 🥺

Reply:
not yet :(